### DISCUSSION TITLE
pipeline with redis cluster

### DISCUSSION THREAD

**Discussion by annahec** [redis/go-redis#2035] [Q&A]
I try to use pipeline with redis cluster client and get sometimes MOVED errors. Using go-redis ver v8.11.4. 
Is this supposed to happen or am I doing something wrong. 
Initialized the pipeline like this: client.Pipeline() (client is of type ClusterClient).
Thanks!

**Comment by vmihailenco** [comment]
Please provide a program that reproduces the problem.

**Reply by annahec** [reply]
This is the code:
var marshalledProps = make(map[string]interface{}, 0)
...
for _, assetInfo := range assetBulk {
...
  err := redisClient.pipe.HSet(redisClient.ctx, assetInfo.AssetId, marshalledProps).Err()
}
res, err := redisClient.pipe.Exec(redisClient.ctx)

I printed res when the problem was reproduced and this is what I got:
res= [hset ae02b4ed7b6c48dea40ac294348db599 deployment "c34ac53c" id "ae02b4ed7b6c48dea40ac294348db599" last_nbt_report_time "1647355942" tenant "6070090b-9336-401c-bf34-f8ed64f52c9b": MOVED 11318 elastic-db-redis-0002-001.elastic-db-redis.nsz0j2.use1.cache.amazonaws.com:6379]

This sometimes happen but not everytime.
What could be the problem?
